superset/migrations/versions/b8d3a24d9131_new_dataset_models.py

```diff
@@ -321,10 +321,14 @@ def after_insert(target: SqlaTable) -> None:  # pylint: disable=too-many-locals
     # table names
     database = (
         target.database
-        or session.query(Database).filter_by(id=target.database_id).one()
+        or session.query(Database).filter_by(id=target.database_id).first()
+    )
+    engine: Optional[Engine] = None
+    if database:
+        engine = database.get_sqla_engine(schema=target.schema)
+    conditional_quote = (
+        engine.dialect.identifier_preparer.quote if engine else lambda x: x
```

Comment:
I think we should just skip the dataset if it has no database... it seems that this is something that shouldn't happen, and a dataset without an associated database is useless, no? Or am I missing something?